server/src/main/java/org/apache/druid/server/coordinator/DruidCoordinatorRuntimeParams.java

```diff
@@ -56,11 +64,50 @@
     return segmentsSet;
   }
 
+  /**
+   * Creates a {@link Set} by choosing used segments from each datasource in a round robin fashion.
+   * The order of segments picked for a datasource is dependent on the iteration order of
+   * {@link ImmutableDruidDataSource#getSegments()}. So if it is desired to have newer segments
+   * be picked up before the older ones, the {@link Collection} returned by {@link ImmutableDruidDataSource#getSegments()}
+   * needs to be setup accordingly.
+   */
+  private static Set<DataSegment> createUsedSegmentSetInRoundRobinFashion(Map<String, ImmutableDruidDataSource> dataSourcesWithAllUsedSegments)
```

Comment:
It would be nice to have some symmetry between this method and `createUsedSegmentsSet`. Maybe they could both accept a DataSourcesSnapshot and both return a Set. Unifying them under a LoadSegmentStrategy interface as suggested by another comment might be a good way to do that.

What do you think?

Reply:
Good idea, @gianm . Will refactor and push a new commit.